packages/mui-base/src/Progress/Root/ProgressContext.tsx

```diff
@@ -0,0 +1,25 @@
+'use client';
+import * as React from 'react';
+import { ProgressContextValue } from './ProgressRoot.types';
+
+export interface ProgressProviderProps {
+  value: ProgressContextValue;
+  children: React.ReactNode;
+}
+
+/**
+ * @ignore - internal component.
+ */
+export const ProgressContext = React.createContext<ProgressContextValue | undefined>(undefined);
```

Comment:
It's possible to use the same name for the const and its type:


```suggestion
export const ProgressContext = React.createContext<ProgressContext | undefined>(undefined);
```

Reply:
@michaldudak How can this work? 🤔 I get an error...
All our other components have `XXXContextValue` as the type name, e.g. `export const PopoverContext = React.createContext<PopoverRootContextValue | null>(null)`

Reply:
OK, let's leave it as it is. I'm trying out a different approach in https://github.com/mui/base-ui/pull/468. If we decide to adopt it, I'll change all the occurences.